### PR TITLE
Fixed pagination to SCA table

### PR DESCRIPTION
**Pull request by chantal-kelm**
### Description

This pull request adds pagination to the table used in the SCA section. This refactor required new endpoints, or new endpoints options (distinct and select) from the Wazuh API.
 
### Issues Resolved

- #4314

### CODE REVIEW

**Comment by Machi3mfl** [comment]
## Researching the current issue

Researching the behavior of the `search bar` and `table` in the `SCA > Inventory` view.
1. The search bar builds the suggestions list calling to the `/sca/AGENT_ID/checks/POLICY_ID}` endpoint retrieving all the checks by policy and making a distinct manually in the `loadScaPolicy` and `buildSuggestionSearchBar` methods.
2. The view uses de `TableWzAPI` to render the table. 

### Steps

- Use the SCA endpoint with the distinct params to prevent requesting all the data needlessly.
- Improve the `inventory.tsx` file because have too much logic and is incomprehensible. It's better to create more files (components) to make the component more readable and reusable.

**Comment by Machi3mfl** [comment]
I found and resolved an error in the `Vulnerabilities > Inventory.` This error was generated by the removed `beautifyDate` function. 

Selection_067.

File: public/components/agents/vuls/inventory.tsx [hunk 1]
```diff
@@ -134,61 +161,53 @@ export class Inventory extends Component {
...
+      </EuiPage>
+    );
   }
 
   // This method was created because Wazuh API returns 1970-01-01T00:00:00Z dates or undefined ones
-  // when vulnerability module is not configured
-  // its meant to render nothing when such date is received
-  beautifyDate(date?: string) {
-    return date && !['1970-01-01T00:00:00Z', '-'].includes(date) ? formatUIDate(date) : '-';
-  }
```
Comment (Desvelao) [inline public/components/agents/vuls/inventory.tsx:180]: **question**: why is it removed?

Comment (Machi3mfl) [reply public/components/agents/vuls/inventory.tsx:180]: Solved

File: public/components/common/tables/table-wz-api.tsx [hunk 2]
```diff
@@ -32,36 +32,38 @@ import { getErrorOrchestrator } from '../../../react-services/common-services';
...
+  title?: string;
+  description?: string;
+  downloadCsv?: boolean;
+  searchTable?: boolean;
+  endpoint: string;
+  buttonOptions?: CustomFilterButton[];
+  onFiltersChange?: Function;
+  showReload?: boolean;
+  searchBarProps?: any;
+  reload?: any;
```
Comment (AlexRuiz7) [inline public/components/common/tables/table-wz-api.tsx:54]: Isn't it boolean?

File: public/components/agents/sca/inventory/lib/api-request.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { WzRequest } from '../../../../../react-services/wz-request';
+
+export async function getFilterValues(
+  field,
+  value,
+  agentId,
+  policyId,
+  filters = {},
+  format = (item) => item
```
Comment (AlexRuiz7) [inline public/components/agents/sca/inventory/lib/api-request.ts:9]: Types?

File: public/components/agents/vuls/inventory.tsx [hunk 4]
```diff
@@ -134,61 +168,53 @@ export class Inventory extends Component {
...
+  // when vulne_isMoun
```
Comment (AlexRuiz7) [inline public/components/agents/vuls/inventory.tsx:208]: Leftover comments

File: public/components/agents/sca/inventory/lib/api-request.ts (hunk 3 above)
Comment (Machi3mfl) [reply public/components/agents/sca/inventory/lib/api-request.ts]: Solved in the last commit

File: public/components/agents/vuls/inventory.tsx (hunk 4 above)
Comment (Machi3mfl) [reply public/components/agents/vuls/inventory.tsx]: Solved in the last commit

**Review by Mayons95 — CHANGES_REQUESTED** [review]
After testing this feature, we can identify that the search bar has some problems.

image

Here I applied a tittle filter and it's ignored by the filter bar.
@Machi3mfl There is a possibility to check this ?

**Comment by Mayons95** [comment]
After reviewing the functionallity with @Machi3mfl we found that the environment API was outdated, after download the changes we could check that the filter bar it's working fine.
I left here the evidence

<details>

Screenshot from 2022-10-19 10-42-57


</details>

**Review by Mayons95 — APPROVED** [review]
LGTM
TR: :heavy_check_mark:

**Review by AlexRuiz7 — APPROVED** [review]
LGTM!

**Pull request merged**